### ISSUE TITLE
bug: Just packages do not work with NodeJS' native ESM implementation

### ISSUE THREAD

**Issue by JakeChampion** [angus-c/just#348]
The extension that is used in Just packages for their esm files is `.esm.js`, this unfortunately does not work for NodeJS' native esm implementation, the extension needs to be `.mjs` for NodeJS to run the file in ESM mode.

Right now when using NodeJS' native esm implementation, an error will be thrown when using `import` on any Just packages:
```
(node:55016) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
/tmp/just-esm-example/node_modules/just-camel-case/index.esm.js:40
export {stringCamelCase as default};
^^^^^^

SyntaxError: Unexpected token 'export'
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1031:15)
    at Module._compile (node:internal/modules/cjs/loader:1065:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:196:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:183:25)
    at async Loader.import (node:internal/modules/esm/loader:178:24)
```

_Originally posted by @JakeChampion in https://github.com/angus-c/just/pull/315#r742839102_

**Comment by angus-c** [comment]
Yep, I'm planning to rewrite all packages to have just one mjs module (and no cjs). Hope to get to it soon.

**Comment by PuruVJ** [comment]
@angus-c would you mind if I make PR for this, if you haven't begun work on it yet? I got a lot of free time at the moment 😅😅

**Comment by PuruVJ** [comment]
See #349

**Comment by PuruVJ** [comment]
Also @JakeChampion in the meanwhile, you can use [all-of-just](https://www.npmjs.com/package/all-of-just). All the Just libraries are bundled in it(still tree-shakeable) , so no issues regarding Node incompatibility. 

Course, I'm plugging my own package here 😅, so when the ESM rewrite of Just is complete, you can move back to it

**Comment by angus-c** [comment]
See WIP PR #345

**Comment by angus-c** [comment]
Should be fixed by #359

**Comment by JakeChampion** [comment]
Great work, thanks @angus-c ☺️
